examples/linear_deformation.py

```diff
@@ -0,0 +1,1187 @@
+
+            grad(S)(alpha) = K alpha
+        """
+#        return self.domain.element([self._kernel_op(np.asarray(a).reshape(-1))
+#                                    for a in alphas])
+        pass
+
+    def _gradient_ft(self, alphas):
+        """Return the gradient at ``alphas``.
+
+        The gradient of the functional is given by
+
+            grad(S)(alpha) = K alpha.
+
+        This is used for the n-D case: control grid = image grid.
+        """
+        ft_momenta = vectorial_ft_shape_op(alphas)
+        return vectorial_ft_shape_op.inverse(self.ft_kernel * ft_momenta)
+#        return (vectorial_ft_op_inverse(ft_displacement) /
+#                self.par_space[0].cell_volume * 2.0 * np.pi)
+
+
+class L2DataMatchingFunctional(Functional):
+
+    """Basic data matching functional using the L2 norm.
+
+    This functional computes::
+
+        1/2 * ||f - g||_2^2
+
+    for a given element ``g``.
+    """
+
+    def __init__(self, space, data):
+        """Initialize a new instance.
+
+        Parameters
+        ----------
+        space : `DiscreteLp` with exponent 2.0
+            Space where the data is matched
+        data : `DiscreteLp` element-like
+            Data which is to be matched
+        """
+        if not (isinstance(space, odl.DiscreteLp) and space.exponent == 2.0):
+            raise ValueError('not an L2 space.')
+        super().__init__(space, linear=False)
+        self.data = self.domain.element(data)
+
+    def _call(self, x):
+        """Return ``self(x)``."""
+        return self.domain.dist(x, self.data)
+
+    def gradient(self, x):
+        """Return the gradient in the point ``x``."""
+        return x - self.data
+
+    def derivative(self, x):
+        """Return the derivative in ``x``."""
+        return self.gradient(x).T
+
+
+# Kernel function for any dimensional
+def gauss_kernel(x, sigma):
+    return np.exp(-x ** 2 / (2 * sigma ** 2))
+
+
+# Kernel function
+def kernel(x):
+    scaled = [xi ** 2 / (2 * sigma ** 2) for xi in x]
+    return np.exp(-sum(scaled))
+
+
+def gaussian_kernel_matrix(grid, sigma):
+    """Return the kernel matrix for Gaussian kernel.
+
+    The Gaussian kernel matrix ``K`` in ``n`` dimensions is defined as::
+
+        k_ij = exp(- |x_i - x_j|^2 / (2 * sigma^2))
+
+    where ``x_i, x_j`` runs through all grid points. The final matrix
+    has size ``N x N``, where ``N`` is the total number of grid points.
+
+    Parameters
+    ----------
+    grid : `RegularGrid`
+        Grid where the control points are defined
+    sigma : `float`
+        Width of the Gaussian kernel
+    """
+    point_arrs = grid.points().T
+    matrices = [parr[:, None] - parr[None, :] for parr in point_arrs]
+    for mat in matrices:
+        mat *= mat
+
+    sq_sum = np.sqrt(np.sum(mat for mat in matrices))
+    kernel_matrix = np.exp(-sq_sum / (2 * sigma ** 2))
+    return kernel_matrix
+
+
+def proj_noise(proj_data_shape, mu=0.0, sigma=0.1):
```

Comment:
Helper functions like this add very little. Instead, simply call `np.random.normal`, or the odl helper [`odl.util.white_noise`](https://github.com/odlgroup/odl/blob/1317d62936a841b2abfb4673b949a8e1b0f06dd7/odl/util/phantom.py#L685-L688).